File: golang/258-add-digits.go
package golang

import "math"

func addDigits(num int) int {
    for int(math.Floor(float64(num / 10))) != 0 {
        sum := 0
        for num > 0 {
            sum += num % 10
            num = int(math.Floor(float64(num / 10)))
        }
        num = sum
    }
    return num
}

File: typescript/258-add-digits.ts
function addDigits(num: number): number {
    while (Math.floor(num / 10) !== 0) {
        let sum = 0
        while (num > 0) {
            sum += num % 10
            num = Math.floor(num / 10)
        }
        num = sum
    }
    return num
};